osu.Game.Rulesets.Taiko/Difficulty/Evaluators/RhythmEvaluator.cs

```diff
@@ -21,27 +21,39 @@ private static double termPenalty(double ratio, int denominator, double power, d
             return -multiplier * Math.Pow(Math.Cos(denominator * Math.PI * ratio), power);
         }
 
+        /// <summary>
+        /// To prevent against infinite ratio values where maps breach regular mapping conditions, we validate the ratio.
+        /// </summary>
+        private static double validateRatio(double ratio)
+        {
+            return double.IsInfinity(ratio) || double.IsNaN(ratio) ? 0 : ratio;
```

Comment:
You can use [double.IsNormal](https://learn.microsoft.com/en-us/dotnet/api/system.double.isnormal?view=net-9.0)

Reply:
copy that